File: QViz.NET/Periscope.QViz/JSON/Action.cs
using System;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for GUI Action Details
	/// </summary>
	public class Action
	{
		/// <summary>
		/// Identifier of the GUI Action
		/// </summary>
		public string actionId;

		/// <summary>
		/// Type of the GUI Action
		/// </summary>
		public string actionType;

		/// <summary>
		/// Name of the Field to perform the GUI Action
		/// </summary>
		public string fieldName;

		/// <summary>
		/// Value of the Field to provide for in the GUI Action
		/// </summary>
		public string fieldValue;

		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;

		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public Action()
		{
			this.actionId = "";
			this.actionType = "";
			this.fieldName = "";
			this.fieldValue = "";
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/Action.java
package com.periscope.qviz.json;

import java.util.Date;

/**
 * JSON Model for GUI Action Details
 */
public class Action {

	/**
	 * Identifier of the GUI Action
	 */
	public String actionId;

	/**
	 * Type of the GUI Action
	 */
	public String actionType;

	/**
	 * Name of the Field to perform the GUI Action
	 */
	public String fieldName;

	/**
	 * Value of the Field to provide for in the GUI Action
	 */
	public String fieldValue;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	public boolean isSensitive;

	public boolean isEncrypted;

	/**
	 * Default Constructor
	 */
	public Action() {
		this.actionId = "";
		this.actionType = "";
		this.fieldName = "";
		this.fieldValue = "";
		this.createdOn = null;
		this.updatedOn = null;
		this.isSensitive = false;
		this.isEncrypted = false;
	}

}
